atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java

```diff
@@ -191,23 +193,9 @@ public SerializableTransaction(
             TableReference tableRef, Iterable<byte[]> rows, BatchColumnRangeSelection columnRangeSelection) {
         Map<byte[], BatchingVisitable<Map.Entry<Cell, byte[]>>> ret =
                 super.getRowsColumnRange(tableRef, rows, columnRangeSelection);
-        return Maps.transformEntries(ret, (row, visitable) -> new BatchingVisitable<Map.Entry<Cell, byte[]>>() {
-            @Override
-            public <K extends Exception> boolean batchAccept(
-                    int batchSize, AbortingVisitor<? super List<Map.Entry<Cell, byte[]>>, K> visitor) throws K {
-                boolean hitEnd = visitable.batchAccept(batchSize, items -> {
-                    if (items.size() < batchSize) {
-                        reachedEndOfColumnRange(tableRef, row, columnRangeSelection);
-                    }
-                    markRowColumnRangeRead(tableRef, row, columnRangeSelection, items);
-                    return visitor.visit(items);
-                });
-                if (hitEnd) {
-                    reachedEndOfColumnRange(tableRef, row, columnRangeSelection);
-                }
-                return hitEnd;
-            }
-        });
+        return KeyedStream.stream(ret)
+                .map((row, visitable) -> wrapWithColumnRangeChecking(tableRef, columnRangeSelection, row, visitable))
+                .collectTo(() -> new TreeMap<>(UnsignedBytes.lexicographicalComparator()));
```

Comment:
[Sanity check]: This collection is consistent with the one in SnapshotTransaction after this PR. I need to validate that this is equivalent before, too.

Is there a way that we can force this coupling? I.e. what if someone in the future changes one of the collectors and not the other?

Reply:
This is equivalent before (see javadoc of `Maps#transformValues`: it takes on properties of the previous map). 

That said, in general, it is not _wrong_ for these to be different types, as long as they satisfy the properties we need (here, mainly that a map can be looked up by value of `byte[]`) so they don't need to be coupled. The tests should fail pretty quickly if that property is broken (since they do perform lookups).

Reply:
I'm convinced this is the same as before, but my concern is that someone depends on this being ordered (i.e. not just a lookup but also an ordering), and that this could be broken in a later PR.

Reply:
I'd be fine with breaking someone on that - the API documentation (see `Transaction`) does not guarantee that behaviour.

Reply:
If we wanted to guarantee that we'd probably return `SortedMap` as interface type as well